src/GameServer.js

```diff
@@ -516,8 +517,16 @@ GameServer.prototype.checkBadWord = function(value) {
 GameServer.prototype.sendChatMessage = function(from, to, message) {
     for (var i = 0, len = this.clients.length; i < len; i++) {
         if (!this.clients[i]) continue;
-        if (!to || to == this.clients[i].playerTracker)
-            this.clients[i].packetHandler.sendPacket(new Packet.ChatMessage(from, message));
+        if (!to || to == this.clients[i].playerTracker){
+            if (this.config.separateChatForTeams && this.gameMode.haveTeams){
+                if (from == null /*server*/ || from.team === this.clients[i].playerTracker.team){

```

Comment:
Remove new line from above an remove comment from /* - You can put this as a new line above. :)